transport/transport_test.go

```diff
@@ -382,6 +382,47 @@ func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, done chan net.Con
 	return tr
 }
 
+// TestInflightStreamClosing ensures that closing in-flight stream
+// sends StreamError to concurrent stream reader.
+func TestInflightStreamClosing(t *testing.T) {
```

Comment:
All fixed.

I can confirm this test fails without the patch.
Our production blocking issue is fixed by this as well.

Thanks!